permissions/src/androidMain/kotlin/dev/icerock/moko/permissions/PermissionsControllerImpl.kt

```diff
@@ -52,28 +53,20 @@ class PermissionsControllerImpl(
         ) { permissions ->
             val isCancelled = permissions.isEmpty()
 
-            val permissionCallback = permissionCallback ?: return@register
-
             if (isCancelled) {
-                permissionCallback.callback.invoke(
-                    Result.failure(RequestCanceledException(permissionCallback.permission))
-                )
+                permissionRequestResultFlow.tryEmit(PermissionRequestResult.Cancelled)
```

Comment:
instead of tryEmit we should use emit i think. in this case if no listeners yet (we in process of activity recreation for example) coroutine will wait result

Reply:
now i see that you use buffered shared flow, so tryEmit will not lost value white activity in process of recreation. so it's ok